zookeeper-server/src/main/java/org/apache/zookeeper/server/ContainerManager.java

```diff
@@ -125,24 +125,29 @@ public void stop() {
      */
     public void checkContainers() throws InterruptedException {
         long minIntervalMs = getMinIntervalMs();
-        for (String containerPath : getCandidates()) {
-            long startMs = Time.currentElapsedTime();
-
-            ByteBuffer path = ByteBuffer.wrap(containerPath.getBytes());
-            Request request = new Request(null, 0, 0, ZooDefs.OpCode.deleteContainer, path, null);
-            try {
-                LOG.info("Attempting to delete candidate container: {}", containerPath);
-                postDeleteRequest(request);
-            } catch (Exception e) {
-                LOG.error("Could not delete container: {}", containerPath, e);
-            }
+        getCandidates().stream()
+                .forEach(containerPath -> {
+                    long startMs = Time.currentElapsedTime();
 
-            long elapsedMs = Time.currentElapsedTime() - startMs;
-            long waitMs = minIntervalMs - elapsedMs;
-            if (waitMs > 0) {
-                Thread.sleep(waitMs);
-            }
-        }
+                    ByteBuffer path = ByteBuffer.wrap(containerPath.getBytes());
+                    Request request = new Request(null, 0, 0, ZooDefs.OpCode.deleteContainer, path, null);
+                    try {
+                        LOG.info("Attempting to delete candidate container: {}", containerPath);
+                        postDeleteRequest(request);
+                    } catch (Exception e) {
+                        LOG.error("Could not delete container: {}", containerPath, e);
+                    }
+
+                    long elapsedMs = Time.currentElapsedTime() - startMs;
+                    long waitMs = minIntervalMs - elapsedMs;
+                    if (waitMs > 0) {
+                        try {
+                            Thread.sleep(waitMs);
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                });
     }
```

Comment:
this change makes the body of `forEach` so fat? just a personal aesthetics.

Reply:
Thanks for the comment! I'll revert back to the original for loop